test/file-handler/file-handler.test.ts

```diff
@@ -0,0 +1,59 @@
+import {FileHandler} from '../../src/file-handler';
+
+const fileHandler = new FileHandler();
+
+describe('FileHandler#extractVersionFromFilename', () => {
+  it('extracts a basic version number from a filename', () => {
+    const filename = '/path/to/file/7.6.1.yaml';
+    const res = fileHandler.extractVersionFromFilename(filename);
+    expect(res).toEqual('7.6.1');
```

Comment:
I'd create a local param for this version number - it'll be much easier to change then. The `7.6.1+rc2` variants and also easily be handled by interpolation.

Reply:
If the number is meaningless in this context, then why not make it completely meaningless, i.e. randomise it?

If you randomise it in each test then your confidence that it's doing the right thing should increase, I would hope.